openedx/features/calendar_sync/views/calendar_sync.py

```diff
@@ -0,0 +1,45 @@
+"""
+Views to toggle Calendar Sync settings for a user on a course
+"""
+
+
+import json
+
+from django.contrib.auth.decorators import login_required
+from django.shortcuts import redirect
+from django.urls import reverse
+from django.utils.decorators import method_decorator
+from django.views.decorators.csrf import ensure_csrf_cookie
+from django.views.generic import View
+from opaque_keys.edx.keys import CourseKey
+
+from openedx.features.calendar_sync.api import (
+    SUBSCRIBE, UNSUBSCRIBE, subscribe_user_to_calendar, unsubscribe_user_to_calendar
+)
+from util.views import ensure_valid_course_key
+
+
+class CalendarSyncView(View):
+    """
+    View for Calendar Sync
+    """
+    @method_decorator(login_required)
+    @method_decorator(ensure_csrf_cookie)
+    @method_decorator(ensure_valid_course_key)
+    def post(self, request, course_id):
+        """
+        Updates the request user's calendar sync subscription status
+
+        Arguments:
+            request: HTTP request
+            course_id (str): string of a course key
+        """
+        course_key = CourseKey.from_string(course_id)
+        json_acceptable_string = request.POST.get('tool_data').replace("'", "\"")
```

Comment:
This feels odd? Why are we not either sending json or just sending bare characters without quoting? Is it because of weird form post syntax?

Reply:
Yeah, so the form data is sent as purely a string. So in our case, we get something like request.POST =
```
{
    'tool_data': "{'toggle_data': 'subscribe'}"
}
```
I felt like this was a good clear way to get it into the state we want.